src/isomorphic/classic/element/ReactElementType.js

```diff
@@ -23,7 +23,7 @@ export type ReactElement = {
   $$typeof: any,
   type: any,
   key: any,
-  ref: any,
+  ref: string | (elem: ?HTMLElement) => void,
```

Comment:
`(elem: ?HTMLElement) => void` isn't right - you can get a component instance passed in (in the case of refs on composite components).

I think we probably want to define this correctly as much as possible and not leave partially correct types on fields (probably fine to leave `any` on to-be type fields).